src/rooms/rooms.gateway.ts

```diff
@@ -19,30 +22,28 @@ export class RoomsGateway implements IRoomsEventPublisher {
   @WebSocketServer()
   server: Server;
 
-  roomCreated(room: Room) {
+  private broadcastEvent<T>(event: string, payload: T) {
     try {
-      const payload = new ResRoomDto(room);
-      void this.server.emit('room:created', payload);
+      void this.server.emit(event, payload);
+
+      const id = (payload as ResRoomDto | ResDeletedRoomDto).id;
+      this.logger.debug(`Event emitted successfully: ${event} - id: ${id}`);
     } catch (err) {
-      this.logger.error(`Failed to emit room created: ${room.id}`, err);
+      this.logger.error(`Failed to emit event: ${event}`, err as Error);
+
+      throw WsError.internalServerError('ROOM_EVENT_FAILED');
```

Comment:
미리 선언해놓은 상수 사용이 안됐네요

```suggestion
      throw WsError.internalServerError(ROOM_ERROR_CODES.ROOM_EVENT_FAILED);
```